CanvasPlusPlayground/Features/Grades/GradesViewModel.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  GradesViewModel.swift
+//  CanvasPlusPlayground
+//
+//  Created by Abdulaziz Albahar on 12/13/24.
+//
+
+import Foundation
+
+@Observable
+class GradesViewModel {
+    var enrollment: Enrollment?
+    
+    let courseId: String
+    
+    init(courseId: String) {
+        self.courseId = courseId
+    }
+    
+    func getEnrollments() async {
+        let request = CanvasRequest.getCourses(enrollmentState: "active")
```

Comment:
while we're here, this can be cleaned up. can pass in courses as a parameter using `CourseManager` as an env obj

Reply:
just realized it should be querying getEnrollments

Reply:
i think the grades are implemented wrong, ill resolve it in a different PR